vsphere/internal/virtualdevice/virtual_machine_cdrom_subresource.go

```diff
@@ -378,6 +384,25 @@ func CdromPostCloneOperation(d *schema.ResourceData, c *govmomi.Client, l object
 	return l, spec, nil
 }
 
+// ValidateCdromConfig takes a CdromSubresource and ensures that the associated configuration is valid
+func (r *CdromSubresource) ValidateCdromConfig() error {
+	log.Printf("[DEBUG] ValidateCdromConfig: Beginning CDROM configuration validation")
+	dsID := r.Get("datastore_id").(string)
+	path := r.Get("path").(string)
+	clientDevice := r.Get("client_device").(bool)
+	if clientDevice {
+		if dsID != "" || path != "" {
+			return fmt.Errorf("ValidateCdromConfig: Cannot have both client_device parameter and ISO file parameters (datastore_id, path) set")
```

Comment:
Can we remove the function name from these error messages? They will be exposed to the user during regular output.